core/src/subgraph/registrar.rs

```diff
@@ -846,14 +846,13 @@ fn reassign_subgraph(
     ops.extend(read_summaries_abort_ops);
 
     ops.push(EntityOperation::AbortUnless {
-        description:
-            "Provided name-deploymentId pair must match an existing, current subgraph version"
-                .to_owned(),
+        description: "Provided name-deploymentId pair must match an existing subgraph version"
+            .to_owned(),
         query: SubgraphEntity::query()
             .filter(EntityFilter::new_in("name", vec![name.clone().to_string()])),
         entity_ids: version_summaries
             .iter()
-            .filter(|version| version.current)
+            .filter(|version| version.deployment_id == hash)
```

Comment:
I am not convinced of the locking logic here: what we want to make sure of is that the assignment is unchanged when we actually do the reassignment operation in the database. I don't think it matters which subgraphs use that deployment, and it's totally fine if some of them stop using this deployment while we reassign, as long as _something_ is still using the deployment.

Reply:
I think something like
```
let current_node_id = lookup the assignment and get its node_id;
ops.push(EntityOperation::AbortUnless {
        description: "Assignment is unchanged".to_owned(),
        query: SubgraphDeploymentAssignmentEntity::query().filter(EntityFilter::And(vec![
            EntityFilter::new_equal("nodeId", current_node_id.to_string()),
            EntityFilter::new_equal("id", hash.clone().to_string()),
        ])),
        entity_ids: vec![hash.clone().to_string()],
});
```

Reply:
Agreed, that makes more sense. I can see where @fordN is coming from though.

Reply:
That's a good point; the subgraph <-> deployment relationship doesn't need to be a concern of the reassigner, but we do need to ensure the deployment doesn't change underneath us.  

Along the same vein,  I don't think we need the version summaries or the version abort operations that come with them. (https://github.com/graphprotocol/graph-node/blob/cad6309086c5754987918a9cae1fb71a9dcb0210/core/src/subgraph/registrar.rs#L844-L846)

Reply:
Yes, I completely agree. The logic should simply be "if that assignment still exists, change the node for the assignment"